### PR TITLE
add reference test for normalize_image_tensor

### PR DESCRIPTION
**Pull request by pmeier**
Addresses https://github.com/pytorch/vision/pull/7113#discussion_r1082155923. 

cc @vfdev-5 @bjuncek

### CODE REVIEW

File: test/prototype_transforms_kernel_infos.py [hunk 1]
```diff
@@ -2232,6 +2232,22 @@ def sample_inputs_normalize_image_tensor():
...
+    sub = torch.Tensor.sub_ if inplace else torch.Tensor.sub
+    return sub(image, mean).div_(std)
+
+
+def reference_inputs_normalize_image_tensor():
+    yield ArgsKwargs(
+        make_image_loader(size=(32, 32), color_space=datapoints.ColorSpace.RGB, extra_dims=[1]),
+        mean=[0.5, 0.5, 0.5],
+        std=[1.0, 1.0, 1.0],
+    )
```
Comment (pmeier) [inline test/prototype_transforms_kernel_infos.py:2248]: This is equivalent to what we had before, but of course we can also expand this.

Comment (NicolasHug) [reply test/prototype_transforms_kernel_infos.py:2248]: What do you mean by "what we had before"?

We have a bunch of tests for the current v1 `normalize()`. I assume we'll still keep these tests around even when we migrate?

Comment (pmeier) [reply test/prototype_transforms_kernel_infos.py:2248]: Yes, sorry. It is equivalent to the test you removed in #7113, i.e.

https://github.com/pytorch/vision/blob/c06d52b1c5f6aee36802661c3ebc6347b97cc59e/test/test_prototype_transforms_functional.py#L1188-L1197

The output type tests are covered by #7118.

Comment (NicolasHug) [reply test/prototype_transforms_kernel_infos.py:2248]: Before merging, should we make sure that all of what is convered in the v1 tests are also covered here? https://github.com/pytorch/vision/blob/01d138d81ae97b846ff63339eeee51f44a1725a2/test/test_transforms.py#L1948-L1999

e.g. the statistical test.

Or is it something we would do later - and in that case, how do we make sure to remember that?

Comment (pmeier) [reply test/prototype_transforms_kernel_infos.py:2248]: As discussed offline, all tests linked above except for the stats one were already covered and even more comprehensive so. I've added the missing stats test in 09b7fb30672e006c9e8758a9b6de8b8224415d70.

File: test/test_prototype_transforms_functional.py [hunk 2]
```diff
@@ -538,6 +543,22 @@ def test_convert_dtype_image_tensor_dtype_and_device(info, args_kwargs, device):
     assert output.device == input.device
 
 
+@pytest.mark.parametrize("device", cpu_and_gpu())
+@pytest.mark.parametrize("num_channels", [3])
```
[the remark this answers is not in the document]
Comment (pmeier) [reply test/test_prototype_transforms_functional.py:547]: Argh, it was 

```suggestion
@pytest.mark.parametrize("num_channels", [1, 3])
```

but changed it for debugging the test.

**Comment by pmeier** [comment]
> (I've seen it fail locally as well, it might be a bit flaky?)

Yeah, we should add some tolerances. What worries me a little is the really high difference albeit only for a single pixel:

```
  FAILED test/test_prototype_transforms_consistency.py::test_call_consistency[ElasticTransform-60] - AssertionError: Tensor image consistency check failed with: 
  
  Tensor-likes are not close!
  
  Mismatched elements: 1 / 318828 (0.0%)
  Greatest absolute difference: 255 at index (2, 0, 159, 72) (up to 1 allowed)
  Greatest relative difference: 1.0 at index (2, 0, 159, 72) (up to 0.1 allowed)
```

I'll look into it.

**Comment by NicolasHug** [comment]
I don't know if you're still working on these `asset_close` utils, but it'd be really nice if they could support something like

```
assert_close(atol, rtol, its_OK_if_that_many_entries_are_off=2)
```

the name is obviously terrible but I hope it illustrates the intent: often tests fail with just one or a few pixels don't satisfy the condition, and that's OK.

**Comment by pmeier** [comment]
> often tests fail with just one or a few pixels don't satisfy the condition, and that's OK.

You are right, but these are very vision centric. Although not publicly documented, we made it really easy to extend `assert_close`. For example, we have 

https://github.com/pytorch/vision/blob/d2d448c71b4cb054d160000a0f63eecad7867bdb/test/prototype_common_utils.py#L51

that we add to `assert_close` so we can do `assert_close(tensor_image, pil_image)`. Plus, it already has support for MAE comparison. That is not 100% what you want, but goes into the same direction: by comparing the MAE, a single pixel in a reasonably sized image makes almost no difference.

**Comment by NicolasHug** [comment]
I wouldn't say they're vision centric; in my scikit-learn days I would have killed for such a feature in the numpy's testing utils. I feel like it's relevant in any scenario when you want to compare an output tensor/array with a reference one.

Re MAE: mean is sensitive to outliers so even if the tensor has a lot of values it can still be quite different - and it's not as easy to hard-code an expected MAE threshold.

**Comment by pmeier** [comment]
Ok, let me clarify:

> fail with just one or a **few pixels** [emphasis mine]

If you are talking about general tensors / arrays I agree that this is not related to vision. I personally only ever encountered this issue with images so I never thought about this being a more general problem. Maybe send an issue to PyTorch core? I'll discuss this with the others if something like that is something we want to support on a public API. I'm currently leaning towards no, since we already have a sh*t ton of parameters and adding more that also tie in the actual comparison won't make it any easier.

If you just want this for `torchvision`, I'm happy to implement a version for us that has this.

**Pull request merged**